reana_workflow_engine_yadage/config.py

```diff
@@ -1,16 +1,26 @@
 # -*- coding: utf-8 -*-
 #
 # This file is part of REANA.
-# Copyright (C) 2017, 2018 CERN.
+# Copyright (C) 2017-2021 CERN.
 #
 # REANA is free software; you can redistribute it and/or modify it
 # under the terms of the MIT License; see LICENSE file for more details.
 
 """REANA Workflow Engine Yadage config."""
 
 import os
+from enum import IntEnum
 
 MOUNT_CVMFS = os.getenv("REANA_MOUNT_CVMFS", "false")
 
 LOGGING_MODULE = "reana-workflow-engine-yadage"
-"""REANA Workflow Engine Yadage logging module."""
+
+WORKFLOW_TRACKING_UPDATE_INTERVAL_SECONDS = 15
+
+
+class WorkflowRunStatus(IntEnum):
```

Comment:
I don't think it can be any more dangerous than it is already. In most of the engines (Snakemake, CWL, Serial), constant numbers (or some local variables) are used to define workflows statuses. In this PR, it is pretty much the same logic as with other engines just implemented with `Enum`.

More on "danger", if the list grows (or changes) we will need to update each engine separately anyway. And, I think this is actually a good thing (opinion below).

*opinion*:  Linking everything together via `reana-commons` makes our components dependant on it, basically, making REANA a monolithic software. It is fine to be dependant on a package when it has a single purpose (`kombu` for MQ as an example) but `reana-commons` is not a single purpose package, it is a collection of different "leftovers" (engines specific code, publishing, disk_quotas, etc.) from other repositories that should actually be in their corresponding repositories (`r-workflow-controller`, `r-engines`, etc.). This is worth a separate discussion :) The point here is that I want to keep as much stuff away as possible from `reana-commons` because I think it is making more harm than good.